modules/backup/variables.tf

```diff
@@ -1,6 +1,11 @@
 variable "resource_name_prefix" {
   description = "Resource name prefix used for tagging and naming AWS resources"
   type        = string
+
+  validation {
+    condition     = can(regex("^[a-zA-Z0-9_-]+$", var.resource_name_prefix))
```

Comment:
I'm not sure but I think that `_` wasn't supported by all resources.

Reply:
I will remove it just in case.

Reply:
Can I start the name with `-`?

Reply:
Yes, it could start with `-`. I have changed it - now it cannot start with `-` but can end with it.